packages/datadog_flutter_plugin/lib/src/helpers.dart

```diff
@@ -40,11 +43,16 @@ void wrap(
   try {
     var result = call();
     if (result is Future) {
-      result.catchError((dynamic e, StackTrace st) =>
-          _handleError(e, st, methodName, logger, attributes));
+      result.catchError((dynamic e, StackTrace st) {
+        if (!_handleError(e, st, methodName, logger, attributes)) {
+          throw e;
```

Comment:
should it be still `throw e` here and not `rethrow`? All the cases below with `_handleError` calls are using `rethrow`.

Reply:
Okay updated. I thought `catchError` worked like `then`, which will preserve the stacktrace if you throw synchronously. It turns out, no, you should supply a `test` function to `catchError` that returns true if you will handle the exception, and passes it on to the Future otherwise.

Thanks for pointing this out!